From 1a349882968763c4b523e20230788f8ca987b5c9 Mon Sep 17 00:00:00 2001
From: Ignacio Casal Quinteiro <qignacio@amazon.com>
Date: Thu, 20 Jan 2022 11:40:06 +0100
Subject: [PATCH] _g_stat_size: return goffset

Otherwise on windows we would be capped at 32bit off_t.
---
 gio/glocalfileinfo.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/glocalfileinfo.h b/gio/glocalfileinfo.h
index e04e921f2..5ca0d9be3 100644
--- a/gio/glocalfileinfo.h
+++ b/gio/glocalfileinfo.h
@@ -297,7 +297,7 @@ inline static guint32   _g_stat_nlink     (const GLocalFileStat *buf) { return b
 #endif
 inline static dev_t     _g_stat_dev       (const GLocalFileStat *buf) { return buf->st_dev; }
 inline static ino_t     _g_stat_ino       (const GLocalFileStat *buf) { return buf->st_ino; }
-inline static off_t     _g_stat_size      (const GLocalFileStat *buf) { return buf->st_size; }
+inline static goffset   _g_stat_size      (const GLocalFileStat *buf) { return buf->st_size; }
 
 #ifndef G_OS_WIN32
 inline static uid_t     _g_stat_uid       (const GLocalFileStat *buf) { return buf->st_uid; }
-- 
2.25.1

